### PR TITLE
Remove leftover strategies and transfer others to remote worker

### PR DESCRIPTION
**Pull request by MTsatsarov**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/591

Related pull request https://github.com/SoftUni-Internal/OpenJudgeSystem/pull/721

Removing from the code the 4 leftover strategies that do not have submission types.
For the others, the submissions types are removed, and the strategies are added to the DisabledLocalWorkerStrategies set.
For the PythonUnitTests strategy should be tested that it is working correctly on the remote worker.

### CODE REVIEW

**Review by mishogv — DISMISSED** [review]

File: OJS.Workers.Common/ExecutionStrategiesConstants.cs [hunk 1]
```diff
@@ -240,6 +241,16 @@ public static class NameMappings
...
+
+                // Ruby
+                ExecutionStrategyType.RubyExecutionStrategy,
+
+                // CSharp
+                ExecutionStrategyType.CSharpProjectTestsExecutionStrategy,
+                ExecutionStrategyType.CSharpUnitTestsExecutionStrategy,
+
+                // Solidity
+                ExecutionStrategyType.SolidityCompileDeployAndRunUnitTestsExecutionStrategy,
```
Comment (gogo4ds) [inline OJS.Workers.Common/ExecutionStrategiesConstants.cs:253]: This does not work on remote worker, shouldn't be removed also

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: OJS.Workers.SubmissionProcessors/Helpers/SubmissionProcessorHelper.cs [hunk 2]
```diff
@@ -381,35 +341,6 @@ public static IExecutionStrategy CreateExecutionStrategy(ExecutionStrategyType t
...
-                case ExecutionStrategyType.SolidityCompileDeployAndRunUnitTestsExecutionStrategy:
-                    executionStrategy = new SolidityCompileDeployAndRunUnitTestsExecutionStrategy(
-                        GetCompilerPath,
-                        processExecutorFactory,
-                        Settings.NodeJsExecutablePath,
-                        Settings.GanacheCliNodeExecutablePath,
-                        Settings.TruffleCliNodeExecutablePath,
-                        int.Parse(submissionProcessorIdentifier),
-                        Settings.SolidityBaseTimeUsedInMilliseconds,
-                        Settings.SolidityBaseMemoryUsedInBytes);
```
Comment (gogo4ds) [inline OJS.Workers.SubmissionProcessors/Helpers/SubmissionProcessorHelper.cs:393]: These settings should be removed if not used. For the other strategies as well.

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: OJS.Workers.SubmissionProcessors/Helpers/SubmissionProcessorHelper.cs (hunk 2 above)
Comment (MTsatsarov) [reply OJS.Workers.SubmissionProcessors/Helpers/SubmissionProcessorHelper.cs:393]: For the solidity they are removed , but for the other strategies it seems that the settings are in use

Comment (gogo4ds) [reply OJS.Workers.SubmissionProcessors/Helpers/SubmissionProcessorHelper.cs:393]: Where are they used? Investigate the usage and see if they are really used. For example SolidityBaseMemoryUsedInBytes is not removed from Settings

Comment (MTsatsarov) [reply OJS.Workers.SubmissionProcessors/Helpers/SubmissionProcessorHelper.cs:393]: DotNet time and memory settings are used in other DotNet strategies,
NUnit is used in CSharpUnitTests strategy,
MsBuild time and memory settings are used in other CSharp strategies,
PhpCli time and memory settings are used in PhpCli and PhpCgi strategies

 Settings.MySqlSysDbConnectionString, Settings.MySqlRestrictedUserId,Settings.MySqlRestrictedUserPassword, these are used in the MySQL strategies.

I thought that i removed the memory and time settings for the solidity, but apparently i forgot, so  will remove them now

Comment (gogo4ds) [reply OJS.Workers.SubmissionProcessors/Helpers/SubmissionProcessorHelper.cs:393]: There is `GanacheCliDefaultPortNumber`. Can you remove it as well

File: OJS.Workers.SubmissionProcessors/Workers/LocalSubmissionWorker.cs [hunk 3]
```diff
@@ -8,8 +8,9 @@ public class LocalSubmissionWorker
     {
         private readonly int portNumber;
 
-        public LocalSubmissionWorker(int portNumber)
-            => this.portNumber = portNumber;
+        public LocalSubmissionWorker()
+        {
+        }
```
Comment (gogo4ds) [inline OJS.Workers.SubmissionProcessors/Workers/LocalSubmissionWorker.cs:13]: We don't need to declare the constructor

File: OJS.Workers.SubmissionProcessors/Workers/LocalSubmissionWorker.cs [hunk 4]
```diff
@@ -8,8 +8,9 @@ public class LocalSubmissionWorker
     {
         private readonly int portNumber;
```
Comment (gogo4ds) [inline OJS.Workers.SubmissionProcessors/Workers/LocalSubmissionWorker.cs:9]: This is not set anywhere, it should be removed

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: OJS.Workers.SubmissionProcessors/SubmissionExecutor.cs [hunk 5]
```diff
@@ -11,6 +11,10 @@ public class SubmissionExecutor : ISubmissionExecutor
     {
         private readonly string submissionProcessorIdentifier;
 
+        public SubmissionExecutor()
+        {
+        }
```
Comment (mishogv) [inline OJS.Workers.SubmissionProcessors/SubmissionExecutor.cs:16]: Why is this needed

**Review by mishogv — CHANGES_REQUESTED** [review]

**Comment by mishogv** [comment]
Please note that there can be some dependencies for strategies installed in Judge_Linux and it is not bad to be refactored as well

File: OJS.Workers.SubmissionProcessors/Workers/LocalSubmissionWorker.cs (hunk 3 above)
Comment (MTsatsarov) [reply OJS.Workers.SubmissionProcessors/Workers/LocalSubmissionWorker.cs]: It was removed

File: OJS.Workers.SubmissionProcessors/Workers/LocalSubmissionWorker.cs (hunk 4 above)
Comment (MTsatsarov) [reply OJS.Workers.SubmissionProcessors/Workers/LocalSubmissionWorker.cs]: It was removed

**Review by gogo4ds — DISMISSED** [review]

**Review by mishogv — DISMISSED** [review]

**Review by gogo4ds — APPROVED** [review]
